plugins/org.jkiss.dbeaver.ext.cubrid/src/org/jkiss/dbeaver/ext/cubrid/model/CubridDataSource.java

```diff
@@ -68,50 +76,127 @@ public GenericTableBase findTable(
-        if (schema.length > 1) {
-            CubridUser user = (CubridUser) this.getSchema(schema[0].toUpperCase());
-            return user.getTable(monitor, schema[1]);
+            throws DBException {
+        if (schemaName != null) {
+            return this.getSchema(schemaName).getTable(monitor, tableName);
+        } else {
+            String[] schemas = tableName.split("\\.");
+            if (schemas.length > 1) {
+                return this.getSchema(schemas[0].toUpperCase()).getTable(monitor, schemas[1]);
+            } else {
+                for (GenericSchema schema : this.getCubridUsers(monitor)) {
+                    GenericTableBase table = schema.getTable(monitor, tableName);
+                    if (table != null) {
+                        return table;
+                    }
+                }
+            }
         }
         return null;
     }
 
     @NotNull
-    public CubridMetaModel getMetaModel()
-    {
+    public CubridMetaModel getMetaModel() {
         return metaModel;
     }
 
+    @NotNull
+    public Collection<CubridCharset> getCharsets() {
+        return charsets;
+    }
+
+    @NotNull
+    public CubridCollation getCollation(String name) {
+        return collations.get(name);
+    }
+
+    @NotNull
     @Override
-    public Collection<? extends DBSDataType> getDataTypes(DBRProgressMonitor monitor) throws DBException
-    {
+    public Collection<? extends DBSDataType> getDataTypes(@NotNull DBRProgressMonitor monitor) throws DBException {
         Map<String, DBSDataType> types = new HashMap<>();
         for (DBSDataType dataType : super.getDataTypes(monitor)) {
             types.put(dataType.getName(), dataType);
         }
         return types.values();
     }
 
-    public CubridObjectContainer getObjectContainer()
-    {
-        return structureContainer;
+    @Nullable
+    public CubridCharset getCharset(@NotNull String name) {
+        for (CubridCharset charset : charsets) {
+            if (charset.getName().equals(name)) {
+                return charset;
+            }
+        }
+        return null;
+    }
+
+    @NotNull
+    public ArrayList<String> getCollations() {
+        ArrayList<String> collationList = new ArrayList<String>(collations.keySet());
+        return collationList;
+    }
+
+    public void loadCharsets(@NotNull DBRProgressMonitor monitor) throws DBException {
+        charsets = new ArrayList<>();
+        try (JDBCSession session = DBUtils.openMetaSession(monitor, container, "Load charsets")) {
+            String sql = "select * from db_charset";
+            final JDBCPreparedStatement dbStat = session.prepareStatement(sql);
+            JDBCResultSet dbResult = dbStat.executeQuery();
+            try {
+                while (dbResult.next()) {
+                    CubridCharset charset = new CubridCharset(this, dbResult);
+                    charsets.add(charset);
+	            }
+	        } finally {
+	             dbStat.close();

```

Comment:
We are always using try catch with resources for statements.

```java
try (JDBCResultSet dbResult = dbStat.executeQuery()) {
}
```

Please fix it.

Reply:
I have fixed this.